tests/tools/test_utils.py

```diff
@@ -28,7 +28,7 @@ def test_persist_execution_report():
             assert full_path.suffix == ".json"
             assert "2000-01-01-000000" in full_path.name
 
-            mocked_json_dumps.called_once_with(report_data)
+            mocked_json_dumps.assert_called_once_with(report_data, sort_keys=True, indent=4)
```

Comment:
This seems to be part of a different set of changes?

Reply:
Technically correct - it fixes this unit test. Python 3.12's `unittest.mock` errors here as it's more strict. I can move it out to a different PR if you'd like.

Reply:
Or make it a separate commit and don't squash it, also fine. I think it is good to have the `TargetPath` change set be as clean as possible.